File: java/src/main/java/com/carepet/model/OwnerDAO.java
package com.carepet.model;

import com.datastax.oss.driver.api.mapper.annotations.Dao;
import com.datastax.oss.driver.api.mapper.annotations.Insert;
import com.datastax.oss.driver.api.mapper.annotations.Select;
import com.datastax.oss.driver.api.mapper.annotations.Update;

import java.sql.Date;
import java.time.Instant;
import java.util.UUID;

@Dao
public interface OwnerDAO {
    @Insert
    void create(Owner owner);

    @Update
    void update(Owner owner);

    @Select
    Owner get(UUID id);

    // @Select
    // List<Pet> getOwnerPets(UUID ownerId);
}

File: csharp/Model/OwnerDAO.cs
using System;
using Cassandra;
using Cassandra.Mapping;

namespace CarePet.Model
{
    public class OwnerDAO
    {
        private readonly IMapper _mapper;

        public OwnerDAO(ISession session)
        {
            _mapper = new Cassandra.Mapping.Mapper(session);
        }

        public void Create(Owner owner)
        {
            _mapper.Insert(owner);
        }

        public void Update(Owner owner)
        {
            _mapper.Update(owner);
        }

        public Owner Get(Guid id)
        {
            return _mapper.FirstOrDefault<Owner>("WHERE owner_id = ?", id);
        }
    }
}
